File: gamelike/item.pl
%% Gamelike
%%
%% (c) 2018 Lyndon Tremblay
%%

:- module(item,[default/1]).

default(I):-
	dict_create(I, item, [name:"Test", weight:1]).

equippable(_):-
	true.

consumable(_):-
	true.

useable(_):-
	true.

sellable(_{cost:N}):-
	N > 0.

item(equip, sword(weight:20, cost:100)).
item(food, potion(weight:2)).
item(food, apple(weight:1)).

%item(Name,Fields):-
%	default(Name).

File: nova/gamelike/item.ml
(*
  Gamelike
  (c) 2019 Lyndon Tremblay
 *)

(* module type Item = sig
 *   type t
 *   val canEquip: t -> bool
 *   val canConsume: t -> bool
 *   val canUse: t -> bool
 *   val canSell: t -> bool
 * end *)
                 
type kind = Material | Equip | Food
type t = {
    kind: kind;
    element: Mob.element;
    name: string option;
    weight: int;
    cost: int;
  }

let empty = {kind= Material;
             element= Mob.Earth;
             name= None;
             weight= 1;
             cost= 1}
           
(* let randomOfKind k = function
 *   | Equip ->  *)
